v3/src/data-interactive/handlers/attribute-handler.ts

```diff
@@ -33,7 +34,9 @@ export const diAttributeHandler: DIHandler = {
       return { success: true }
     }).filter(error => !error.success)
     if (attributeErrors.length > 0) return attributeErrors[0]
-
+    if (!metadata && !Array.isArray(values) && 'colormap' in values && values.colormap) {
```

Comment:
This fail case should occur if _any_ attribute includes `colormap` and we can't find `metadata`.
```suggestion
    if (attributeValues.some(attribute => !!attribute.colormap) && !metadata) {
```